src/crimson/osd/ops_executer.h

```diff
@@ -151,6 +151,7 @@ class OpsExecuter : public seastar::enable_lw_shared_from_this<OpsExecuter> {
     virtual ~effect_t() = default;
   };
 
+  Ref<PG> pg; // for the sake of the ObjClass API
```

Comment:
Not sure we really need the sharing behavior here; likely a plain reference would do.

Additionally, the `backend` member can be stripped if we have the PG reference.

Reply:
Stripped the backend and pool_info members. I think the sharing behavior is needed for `OpsExecuter::flush_changes_n_do_ops_effects()`, see https://github.com/ceph/ceph/blob/f435859ec4ee4b5dc9903c209333ddbd17f7e1da/src/crimson/osd/ops_executer.h#L357